src/edu/washington/escience/myria/operator/IDBController.java

```diff
@@ -309,6 +330,31 @@ public final void init(final ImmutableMap<String, Object> execEnvVars) throws Db
     state.init(execEnvVars);
     deltaTuples = new LinkedList<TupleBatch>();
     bufferedIterTBs = new ArrayList<TupleBatch>();
+
+    if (relationKey != null) {
+      if (connectionInfo == null && execEnvVars != null) {
+        connectionInfo = (ConnectionInfo) execEnvVars.get(MyriaConstants.EXEC_ENV_VAR_DATABASE_CONN_INFO);
+      }
+      if (connectionInfo == null) {
+        throw new DbException("Unable to instantiate DbInsert: connection information unknown");
+      }
+      if (connectionInfo instanceof SQLiteInfo) {
+        /* Set WAL in the beginning. */
+        final File dbFile = new File(((SQLiteInfo) connectionInfo).getDatabaseFilename());
+        SQLiteConnection conn = new SQLiteConnection(dbFile);
+        try {
+          conn.open(true);
+          conn.exec("PRAGMA journal_mode=WAL;");
```

Comment:
I don't see the need for this. WAL mode is persistent, so it should only need to be supplied at DB creation.